test/test_s3.py

```diff
@@ -240,6 +242,7 @@ def test_put_object_multiple_times(self):
                 on_headers=self._on_request_headers,
                 on_body=self._on_request_body)
             finished_futures.append(s3_request.finished_future)
+            del s3_request
```

Comment:
why this is not released with as it goes out of scope?

Reply:
I had a discussion with Mike and Python has a weird behavior where it is not guaranteed that something will delete when it goes out of scope.